importexport/musicxml/exportxml.cpp

```diff
@@ -2923,30 +2923,43 @@ static void writeBeam(XmlWriter& xml, ChordRest* const cr, Beam* const b)
       int blp = -1; // beam level previous chord
       int blc = -1; // beam level current chord
       int bln = -1; // beam level next chord
+      Beam::Mode bmc = Beam::Mode::AUTO; // beam mode current chord
+      Beam::Mode bmn = Beam::Mode::AUTO; // beam mode next chord
       // find beam level previous chord
       for (int i = idx - 1; blp == -1 && i >= 0; --i) {
             const auto crst = elements[i];
             if (crst->isChord())
                   blp = toChord(crst)->beams();
             }
       // find beam level current chord
-      if (cr->isChord())
+      if (cr->isChord()) {
             blc = toChord(cr)->beams();
+            bmc = toChord(cr)->beamMode();
+            }
       // find beam level next chord
       for (int i = idx + 1; bln == -1 && i < elements.size(); ++i) {
             const auto crst = elements[i];
-            if (crst->isChord())
+            if (crst->isChord()) {
                   bln = toChord(crst)->beams();
+                  bmn = toChord(crst)->beamMode();
+                  }
             }
       // find beam type and write
       for (int i = 1; i <= blc; ++i) {
             QString text;
-            if (blp < i && bln >= i) text = "begin";
+            // TODO: correctly handle Beam::Mode::AUTO
+            // when equivalent to BEGIN32 or BEGIN64
+            if ((blp < i && bln >= i)
+             || bmc == Beam::Mode::BEGIN32 && i > 1
+             || bmc == Beam::Mode::BEGIN64 && i > 2)
```

Comment:
this introduces compiler warning, suggetion to add parens or the `&&` conditions

Reply:
#8554

Reply:
Thanks!

Reply:
Likewise! I've been trying to keep a closer eye on warnings as I compile locally; hopefully no more will slip through!